apps/examples/testcase/le_tc/kernel/itc_semaphore.c

```diff
@@ -0,0 +1,174 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file itc_semaphore.c
+/// @brief Test Case Example for semaphore API
+#include <tinyara/config.h>
+#include <semaphore.h>
+#include <pthread.h>
+#include "tc_internal.h"
+#include <stdio.h>
+#include <sys/types.h>
+
+
+#define LOOP_SIZE CONFIG_ITC_KERN_SEM_LOOP_SIZE
+#define PSHARED 0
+/**
+* @fn                   :itc_semaphore_sem_wait_n_after_sem_destroy
+* @brief                :sem_post, sem_wait
+* @scenario             :init semaphore, then delete semaphore. After that call sem_wait
+* API's covered         :sem_post, sem_wait
+* Preconditions         :Semaphore should be destroyed.
+* Postconditions        :NA
+*/
+static void itc_semaphore_sem_wait_n_after_sem_destroy(void)
```

Comment:
In POSIX description of sem_destroy, below is written.
```
The sem_destroy() function shall destroy the unnamed semaphore indicated by sem. Only a semaphore that was created using sem_init() may be destroyed using sem_destroy(); the effect of calling sem_destroy() with a named semaphore is undefined. The effect of subsequent use of the semaphore sem is undefined until sem is reinitialized by another call to sem_init().

It is safe to destroy an initialized semaphore upon which no threads are currently blocked. The effect of destroying a semaphore upon which other threads are currently blocked is undefined.
```

So, we don't support the error on your itc situation (calling sem functions after calling sem_destroy). And we can't say that is wrong. We need to block thest kind TCs.

Reply:
yes agree with your comment,removed this testcase,In Sample code on Linux Ubuntu 14.04, sem_wait returning success,after being destroyed.The behavior is not defined clearly.